File: frontend/src/app/models/gamedataDTOs/GameDataDTO.ts
import { CharacterClass } from "../../enums/CharacterClass";
import { StatType } from "../../enums/StatType";
import { WisdomGroupType } from "../../enums/WisdomGroupType";
import { WisdomSkillType } from "../../enums/WisdomSkillType";
import { BuffDefinitionDTO } from "./BuffDefinitionDTO";
import { ClassStatsDTO } from "./ClassStatsDTO";
import { CollectorBagCategoryBonusDefinitionDTO } from "./CollectorBagCategoryBonusDefinitionDTO";
import { DragonStoneDefinitionDTO } from "./DragonStoneDefinitionDTO";
import { EnchantmentDTO } from "./EnchantmentDTO";
import { EssenceDefinitionDTO } from "./EssenceDefinitionDTO";
import { GemDefinitionDTO } from "./GemDefinitionDTO";
import { ItemDefinitionDTO } from "./ItemDefinitionDTO";
import { JewelDefinitionDTO } from "./JewelDefinitionDTO";
import { LevelMultiplierTableDTO } from "./LevelMultiplierTableDTO";
import { PetDefinitionDTO } from "./PetDefinitionDTO";
import { RuneDefinitionDTO } from "./RuneDefinitionDTO";
import { SetDTO } from "./SetDTO";
import { WisdomGroupDefinitionDTO } from "./WisdomGroupDefinitionDTO";
import { WisdomSkillDefinitionDTO } from "./WisdomSkillDefinitionDTO";

export interface GameDataDTO {
    characterClassStats:  Record<string, ClassStatsDTO>;
    items: Record<CharacterClass, Record<string, ItemDefinitionDTO>>;
    sets: Record<string, Record<string, SetDTO>>;
    jewels: Record<string, Record<string, JewelDefinitionDTO>>;
    jewelLimits: Record<string, number>;
    enchantments: Partial<Record<StatType, EnchantmentDTO>>;
    gems: Record<string, GemDefinitionDTO>;
    gemLimits: Record<string, number>;
    runes: Record<string, RuneDefinitionDTO>;
    runeLimits: Record<string, number>;
    dragonStones: Record<string, DragonStoneDefinitionDTO>;
    pets: Record<string, PetDefinitionDTO>;
    essences: Record<string, EssenceDefinitionDTO>;
    tonics: Record<string, BuffDefinitionDTO>;
    physics: Record<string, BuffDefinitionDTO>;
    levelMultiplierTable: LevelMultiplierTableDTO;
    wisdomSkills: Record<WisdomSkillType, WisdomSkillDefinitionDTO>
    wisdomGroups: Record<WisdomGroupType, WisdomGroupDefinitionDTO>
    collectorBagBuffs: CollectorBagCategoryBonusDefinitionDTO[];
}

File: backend/src/main/java/com/langleon/dsobuildsim/gamedata/dto/GameDataDTO.java
package com.langleon.dsobuildsim.gamedata.dto;

import com.langleon.dsobuildsim.buffs.dto.BuffDefinitionDTO;
import com.langleon.dsobuildsim.character.CharacterClass;
import com.langleon.dsobuildsim.collectorbagbonus.dto.definition.CollectorBagCategoryBonusDefinitionDTO;
import com.langleon.dsobuildsim.dragonstones.DragonStoneType;
import com.langleon.dsobuildsim.dragonstones.dto.DragonStoneDefinitionDTO;
import com.langleon.dsobuildsim.gems.enums.GemLimitGroup;
import com.langleon.dsobuildsim.gems.enums.GemType;
import com.langleon.dsobuildsim.items.core.enums.ItemType;
import com.langleon.dsobuildsim.items.dto.ItemDefinitionDTO;
import com.langleon.dsobuildsim.jewels.JewelLimitGroup;
import com.langleon.dsobuildsim.jewels.JewelType;
import com.langleon.dsobuildsim.runes.enums.RuneLimitGroup;
import com.langleon.dsobuildsim.runes.enums.RuneType;
import com.langleon.dsobuildsim.sets.SetType;
import com.langleon.dsobuildsim.wisdomskilltree.dto.definition.WisdomGroupDefinitionDTO;
import com.langleon.dsobuildsim.wisdomskilltree.dto.definition.WisdomSkillDefinitionDTO;
import com.langleon.dsobuildsim.enchantments.dto.EnchantmentDTO;
import com.langleon.dsobuildsim.essences.dto.EssenceDefinitionDTO;
import com.langleon.dsobuildsim.gems.dto.GemDefinitionDTO;
import com.langleon.dsobuildsim.jewels.dto.JewelDefinitionDTO;
import com.langleon.dsobuildsim.pets.dto.PetDefinitionDTO;
import com.langleon.dsobuildsim.runes.dto.RuneDefinitionDTO;
import com.langleon.dsobuildsim.sets.dto.SetDTO;
import com.langleon.dsobuildsim.wisdomskilltree.wisdomgroup.WisdomGroupType;
import com.langleon.dsobuildsim.wisdomskilltree.wisdomskill.WisdomSkillType;

import java.util.List;
import java.util.Map;

public record GameDataDTO (
        Map<CharacterClass, ClassStatsDTO> characterClassStats,
        Map<CharacterClass, Map<ItemType, ItemDefinitionDTO>> items,
        Map<CharacterClass, Map<SetType, SetDTO>> sets,
        Map<CharacterClass, Map<JewelType, JewelDefinitionDTO>> jewels,
        Map<JewelLimitGroup, Integer> jewelLimits,
        List<EnchantmentDTO> enchantments,
        Map<GemType, GemDefinitionDTO> gems,
        Map<GemLimitGroup, Integer> gemLimits,
        Map<RuneType, RuneDefinitionDTO> runes,
        Map<RuneLimitGroup, Integer> runeLimits,
        Map<DragonStoneType, DragonStoneDefinitionDTO> dragonStones,
        List<PetDefinitionDTO> pets,
        List<EssenceDefinitionDTO> essences,
        List<BuffDefinitionDTO> tonics,
        List<BuffDefinitionDTO> physics,
        LevelMultiplierTableDTO levelMultiplierTable,
        Map<WisdomSkillType, WisdomSkillDefinitionDTO> wisdomSkills,
        Map<WisdomGroupType, WisdomGroupDefinitionDTO> wisdomGroups,
        List<CollectorBagCategoryBonusDefinitionDTO> collectorBagBuffs
        )
{
}
